### PR TITLE
`azurerm_kubernetes_cluster`: Expose `automatic_channel_upgrade`

### LINKED ISSUE #10407: Support for auto upgrade channel
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support setting auto-upgrade channel on an AKS cluster
https://docs.microsoft.com/en-us/azure/aks/upgrade-cluster#set-auto-upgrade-channel

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster_node_pool" "example" {
  name                  = "foo"
  vm_size               = "Standard_DS2_v2"
  node_count            = 3
  
  auto_upgrade_channel = "patch"
}
```

### References

* https://docs.microsoft.com/en-us/azure/aks/upgrade-cluster#set-auto-upgrade-channel

### PR DESCRIPTION
**Pull request by brunhil**
Fixes #10407 

=== RUN   TestAccKubernetesCluster_upgradeChannel
=== PAUSE TestAccKubernetesCluster_upgradeChannel
=== CONT  TestAccKubernetesCluster_upgradeChannel
--- PASS: TestAccKubernetesCluster_upgradeChannel (824.85s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers  828.691s

### CODE REVIEW

**Review by shapeofarchitect — APPROVED** [review]

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -580,6 +580,18 @@ func resourceKubernetesCluster() *schema.Resource {
 				},
 			},
 
+			"auto_upgrade_channel": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_cluster_resource.go:583]: would this make more sense as
```suggestion
			"automatic_channel_upgrade": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go (hunk 1 above)
Comment (shapeofarchitect) [reply azurerm/internal/services/containers/kubernetes_cluster_resource.go:583]: yes agree - makes more sense.

**Review by tombuildsstuff — APPROVED** [review]
hey @brunhil 

Thanks for pushing those changes.

Taking a look through on the whole this looks pretty good - but we should remove the `none` option here in favour of inferring that when the field is omitted (since this is Optional and will be unset by default). Since this otherwise looks good and I've got a few other Kubernetes PR's to get in today, I hope you don't mind but I'm going to push a commit to fix up the comments so that we can get this merged.

Thanks!

**Review by jackofallops — APPROVED** [review]
Thanks @brunhil & @tombuildsstuff - LGTM 👍

**Comment by tombuildsstuff** [comment]
The basicVMSS and Upgrade Channel tests look good:

```
$ TF_ACC=1 go test -v ./azurerm/internal/services/containers -run=TestAccKubernetesCluster_upgradeChannel -timeout=60m
=== RUN   TestAccKubernetesCluster_upgradeChannel
=== PAUSE TestAccKubernetesCluster_upgradeChannel
=== CONT  TestAccKubernetesCluster_upgradeChannel
--- PASS: TestAccKubernetesCluster_upgradeChannel (1045.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers	1045.774s
```

**Pull request merged**